File: eecs168/2021spring/lectures/TR/2021.04.20.classes-part-2/Triangle.cpp
//Triangle.cpp
#include "Triangle.h"

Triangle::Triangle()
{
	//Assigning base and height initial values
	m_base = 0;
	m_height = 0;
}

//This method belongs to the Triangle
double Triangle::area()
{
	//base and height's values will be 
	//determined based on which instance/object
	//is doing the calling
	double ans = 0;
	ans = 0.5*m_base*m_height;
	return(ans);
}

double Triangle::getBase()
{
	return(m_base);
}

double Triangle::getHeight()
{
	return(m_height);
}

bool Triangle::setBase(double base)
{
	if(base>0)
	{
		m_base = base;
		return(true);
	}
	else
	{
		return(false);
	}
	
}

bool Triangle::setHeight(double height)
{
	if(height>0)
	{
		m_height = height;
		return(true);
	}
	else
	{
		return(false);
	}
	
}

File: eecs168/2025spring/TR/2025.04.24.classes-part-3/triangle/triangle.py
#triangle.py

class Triangle:
    def __init__(self, base, height):
        print('init called')
        self.base = base
        self.height = height

    def area(self):
        ans = (1/2)*self.base*self.height
        return ans
        
    def __gt__(self, other):
        if self.area() > other.area():
            return True
        else:
            return False

    def __eq__(self, other):
        if self.area() == other.area():
            return True
        else:
            return False
